tests/backup/backup_test.go

```diff
@@ -6514,6 +6514,175 @@ var _ = Describe("{IssueMultipleDeletesForSharedBackup}", func() {
+				}
+
+			}
+		})
+		for _, user := range users {
+			Step(fmt.Sprintf("Register source and destination cluster for backup on %s ", user), func() {
+				log.InfoD("Registering Source and Destination clusters as user : %s and verifying the status", user)
+				ctx, err := backup.GetNonAdminCtx(user, "Password1")
+				log.FailOnError(err, "Fetching %s ctx", user)
+				CreateSourceAndDestClusters(orgID, "", "", ctx)
+				time.Sleep(time.Minute * 2)
+				clusterStatus, clusterUid = Inst().Backup.RegisterBackupClusterNonAdminUser(orgID, SourceClusterName, "", ctx)
+				dash.VerifyFatal(clusterStatus, api.ClusterInfo_StatusInfo_Online, "Verifying backup cluster status")
+			})
+			Step(fmt.Sprintf("Taking backup of applications as %s", user), func() {
+				log.InfoD("Taking backup of applications as user : %s ", user)
+				backupName = "backup1"
+				ctx, err := backup.GetNonAdminCtx(user, "Password1")
+				log.FailOnError(err, "Fetching testuser ctx")
+				err = CreateBackup(backupName, SourceClusterName, userbackupLocationMapping[user], backupLocationUID, []string{bkpNamespaces[0]},
+					nil, orgID, clusterUid, "", "", "", "", ctx)
+				dash.VerifyFatal(err, nil, "Verifying backup creation")
+			})
+		}
+		Step(fmt.Sprintf("Share backup with %s", users[0]), func() {
+			log.InfoD(fmt.Sprintf("Share backup from %s to %s and validate", users[0], users[1]))
+			ctx, err := backup.GetNonAdminCtx(users[0], "Password1")
+
+			// Share backup with the user
+			err = ShareBackup(backupName, nil, []string{users[1]}, RestoreAccess, ctx)
+			log.FailOnError(err, "Failed to share backup %s", backupName)
+		})
+		Step(fmt.Sprintf("validate the backup shared %s is present in user context %s", backupName, users[1]), func() {
+			userBackups, _ := GetAllBackupsForUser(users[1], "Password1")
+			backupCount := 0
+			for _, backup := range userBackups{
+				if backup == backupName {
+					backupCount = backupCount + 1
+				}
+			}
+			dash.VerifyFatal(backupCount, 2, fmt.Sprintf("Validating the shared backup [%s] is present in user context [%s]", backupName, users[1]))
+		})
+		Step(fmt.Sprintf("Share backup with %s", users[1]), func() {
+			log.InfoD(fmt.Sprintf("Share backup from %s to %s and validate", users[1], users[0]))
+			ctx, err := backup.GetNonAdminCtx(users[1], "Password1")
+
+			// Share backup with the user
+			err = ShareBackup(backupName, nil, []string{users[0]}, RestoreAccess, ctx)
+			log.FailOnError(err, "Failed to share backup %s", backupName)
+		})
+		Step(fmt.Sprintf("validate the backup shared %s is present in user context %s", backupName, users[0]), func() {
+			userBackups, _ := GetAllBackupsForUser(users[1], "Password1")
+			backupCount := 0
+			for _, backup := range userBackups{
+				if backup == backupName {
+					backupCount = backupCount + 1
+				}
+			}
+			dash.VerifyFatal(backupCount, 2, fmt.Sprintf("Validating the shared backup [%s] is present in user context [%s]", backupName, users[0]))
+		})
+	})
+	JustAfterEach(func() {
```

Comment:
There is no cleanup for backups and backup location

Reply:
no clean up required is required here as the location is created in user context , and user is wiped out in the clean up